influxdb_iox/src/influxdb_ioxd/http/dml.rs

```diff
@@ -239,7 +239,7 @@ pub trait HttpDrivenDml: ServerType {
             "inserting lines into database",
         );
 
-        let write = DmlWrite::new(tables, DmlMeta::unsequenced(span_ctx));
+        let write = DmlWrite::new(db_name.as_str(), tables, DmlMeta::unsequenced(span_ctx));
```

Comment:
We can avoid converting the `DatabaseName` to a `str` and then [to a `String`](https://github.com/influxdata/influxdb_iox/blob/8180b007aec9cee33276a33a67de002f73d30032/dml/src/lib.rs#L224) straight after by using `&db_name`:

```suggestion
        let write = DmlWrite::new(&db_name, tables, DmlMeta::unsequenced(span_ctx));
```

This works because there is a `From<&DatabaseName> for String` [impl allowing the direct conversion](https://github.com/influxdata/influxdb_iox/blob/8180b007aec9cee33276a33a67de002f73d30032/data_types/src/database_name.rs#L88-L92).

Same thing for the delete path too 👍